lib/airbrake-ruby/filter_chain.rb

```diff
@@ -45,11 +54,12 @@ def initialize(config)
     # @param [#call] filter The filter object (proc, class, module, etc)
     # @return [void]
     def add_filter(filter)
-      return @keys_filters << filter if KEYS_FILTERS.include?(filter.class)
-      return @filters << filter unless filter.to_s.start_with?(LIB_NAMESPACE)
-
-      i = @filters.rindex { |f| f.to_s.start_with?(LIB_NAMESPACE) }
-      @filters.insert(i + 1, filter) if i
+      if filter.to_s.start_with?(LIB_NAMESPACE)
+        @filters.insert(@offset_left, filter)
+        @offset_left += 1
+      else
+        @filters.insert(@filters.size - @offset_right, filter)
+      end
```

Comment:
This is still ugly and unreadable. What is the problem with adding optional `weight` property to filters (with default value 0) and sort by `filter.weight`? It should not be more than 10 lines of code.